### PR TITLE
chore(generate-loader,plugin): add typescript support

### PR DESCRIPTION
**Pull request by dhruvdutt**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

**Comment by dhruvdutt** [comment]
> This one looks good, but the classes should be ts too

Which classes?

**Comment by ev1stensberg** [comment]
@dhruvdutt Think they live in the `generators` package now, sorry.

File: custom_typings/yeoman-environment.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+declare module 'yeoman-environment' {
```
Comment (ev1stensberg) [inline custom_typings/yeoman-environment.d.ts]: Change this folder name to types

File: packages/generate-loader/index.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/// <reference path="../../custom_typings/yeoman-environment.d.ts" />
```
Comment (ev1stensberg) [inline packages/generate-loader/index.ts]: Is this needed? Types could be imported easier

File: packages/generate-plugin/index.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/// <reference path="../../custom_typings/yeoman-environment.d.ts" />
+import { PluginGenerator } from "@webpack-cli/generators/plugin-generator";
```
Comment (ev1stensberg) [inline packages/generate-plugin/index.ts]: Same here

File: packages/webpack-scaffold/index.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import jscodeshift from "jscodeshift";
+
+interface IScaffoldBaseObject {
```
Comment (ev1stensberg) [inline packages/webpack-scaffold/index.ts]: Change basename to inquirer

File: packages/generate-loader/types/Yeoman.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export interface IYeoman {
```
Comment (ev1stensberg) [inline packages/generate-loader/types/Yeoman.ts]: `any` makes use of typescript useless. Debug the types yourself to find the correct ones

File: packages/generate-plugin/types/Yeoman.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export interface IYeoman {
```
Comment (ev1stensberg) [inline packages/generate-plugin/types/Yeoman.ts]: Same here as previous comment

File: packages/generate-loader/types/Yeoman.ts (hunk 5 above)
Comment (dhruvdutt) [reply packages/generate-loader/types/Yeoman.ts]: The methods present in this interface are typed precisely and not using `any`. Where exactly are you pointing?

Comment (ev1stensberg) [reply packages/generate-loader/types/Yeoman.ts]: It was `any` before, maybe I reviewed an old commit

File: packages/generators/addon-generator.ts [hunk 7]
```diff
@@ -27,19 +27,19 @@ import { IScaffoldBaseObject } from "@webpack-cli/webpack-scaffold";
...
 ) {
 	return class AddOnGenerator extends Generator {
-		public props: IScaffoldBaseObject;
+		public props: IInquirerScaffoldObject;
 		private copy: (value: string, index: number, array: string[]) => void;
 		private copyTpl: (value: string, index: number, array: string[]) => void;
 
 		public prompting() {
-			return this.prompt(prompts).then((props: IScaffoldBaseObject) => {
+			return this.prompt(prompts).then((props: IInquirerScaffoldObject) => {
```
Comment (ev1stensberg) [inline packages/generators/addon-generator.ts]: This would return a promise, where's the type decl on that?

**Comment by ev1stensberg** [comment]
@dhruvdutt you'll need to work on this this weekend

**Comment by dhruvdutt** [comment]
@ev1stensberg Updated the change you suggested. :)

**Pull request merged**